In [ ]:
# Charger le dataset
import pandas as pd
df = pd.read_csv('/content/diabetes (1).csv')
#Les premières lignes
df.head()

,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
0,6,148,72,35,0,33.6,0.627,50,1
1,1,85,66,29,0,26.6,0.351,31,0
2,8,183,64,0,0,23.3,0.672,32,1
3,1,89,66,23,94,28.1,0.167,21,0
4,0,137,40,35,168,43.1,2.288,33,1


In [ ]:
# Structure du dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [ ]:
# Nombre de valeurs manquantes
df.isnull().sum()

,0
Pregnancies,0
Glucose,0
BloodPressure,0
SkinThickness,0
Insulin,0
BMI,0
DiabetesPedigreeFunction,0
Age,0
Outcome,0


In [ ]:
# Identifier les zéros
(df == 0).sum()

,0
Pregnancies,111
Glucose,5
BloodPressure,35
SkinThickness,227
Insulin,374
BMI,11
DiabetesPedigreeFunction,0
Age,0
Outcome,500


In [ ]:
# Analyse statistique
df.describe()

,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
count,768.000000,768.000000,768.000000,768.000000,768.000000,768.000000,768.000000,768.000000,768.000000
mean,3.845052,120.894531,69.105469,20.536458,79.799479,31.992578,0.471876,33.240885,0.348958
std,3.369578,31.972618,19.355807,15.952218,115.244002,7.884160,0.331329,11.760232,0.476951
min,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.078000,21.000000,0.000000
25%,1.000000,99.000000,62.000000,0.000000,0.000000,27.300000,0.243750,24.000000,0.000000
50%,3.000000,117.000000,72.000000,23.000000,30.500000,32.000000,0.372500,29.000000,0.000000
75%,6.000000,140.250000,80.000000,32.000000,127.250000,36.600000,0.626250,41.000000,1.000000
max,17.000000,199.000000,122.000000,99.000000,846.000000,67.100000,2.420000,81.000000,1.000000


D'après l'analyse statistique pour ce dataset, il y a 768 entrées pour chaque colonne, ce qui signifie qu'il n'y a pas de valeurs manquantes.

Pour la colonne Glucose, la moyenne est d'environ 120.89 et l'écart type est de 31.97, avec un minimum de 0 et un maximum de 199. Le fait que le minimum de plusieurs colonnes (Glucose, BloodPressure, SkinThickness, Insulin, BMI) soit 0 est une anomalie car ces valeurs ne peuvent pas être nulles dans un contexte biologique sain, et cela pourrait indiquer des valeurs manquantes codées en zéro.

In [ ]:
pip install ydata-profiling

In [ ]:
from ydata_profiling  import ProfileReport
# Générer le Rapport
profile = ProfileReport(df,title="Profiling du Diabète")

#Exporter le rapport en HTML
profile.to_file("rapport_profiling_diabete.html")

Les observations clés tirées de Pandas:

Absence de valeurs manquantes apparentes: La fonction df.isna().sum() a confirmé qu'il n'y a pas de valeurs NaN explicites dans le dataset, toutes les colonnes ayant 768 entrées complètes.
Anomalies dans les valeurs minimales: L'analyse statistique (df.describe()) a révélé que plusieurs colonnes importantes comme Glucose, BloodPressure, SkinThickness, Insulin et BMI ont des valeurs minimales de 0. Biologiquement, ces valeurs ne peuvent pas être nulles (par exemple, une pression artérielle de 0), ce qui suggère fortement que ces zéros représentent en réalité des valeurs manquantes ou non enregistrées.

Les observations clés tirées du rapport de Profilage:


1. Aperçu Général du Dataset:

Le dataset contient 9 variables (colonnes) et 768 observations (lignes).
Il est composé de 8 variables numériques et 1 variable catégorique (Outcome est probablement traitée comme catégorique car elle est binaire : 0 ou 1).
La taille totale en mémoire est de 54,1 Kio.


2. Corrélations :

Prédiction du Diabète (Outcome): La variable Outcome (résultat) montre une corrélation positive la plus forte avec Glucose (0.487). Cela indique que des niveaux de glucose plus élevés sont fortement associés à la présence de diabète.


3. Corrélations Fortes Entres Variables:

Age et Pregnancies: Une forte corrélation est observée entre l'âge et le nombre de grossesses. Cela est attendu et logique, car le nombre de grossesses augmente généralement avec l'âge d'une femme.

Insulin et SkinThickness: Une forte corrélation existe également entre les niveaux d'insuline et l'épaisseur de la peau. Cela peut être lié à la composition corporelle et au métabolisme.

4. Présence de Zéros Biologiquement Impossibles:

Pregnancies: 111 observations (14,5%) ont une valeur de 0. Bien que 0 grossesse soit possible, ces zéros pourraient nécessiter une vérification contextuelle si l'objectif est d'étudier la grossesse active.

BloodPressure: 35 observations (4,6%) ont une valeur de 0. Une pression artérielle de 0 est cliniquement impossible et indique clairement des données manquantes qui ont été encodées en zéro.

SkinThickness: 227 observations (29,6%) ont une valeur de 0. L'épaisseur de la peau ne peut pas être nulle, ce qui indique que près d'un tiers des données pour cette colonne sont manquantes et encodées en zéro.

Insulin: 374 observations (48,7%) ont une valeur de 0. Près de la moitié des données d'insuline sont à zéro, ce qui est biologiquement impossible pour un être vivant et indique des valeurs manquantes.

BMI (Indice de Masse Corporelle): 11 observations (1,4%) ont une valeur de 0. Un IMC de 0 est également impossible et représente des données manquantes.

Implications pour la suite: Ces alertes soulignent la nécessité de traiter ces valeurs de zéro comme des valeurs manquantes pour les colonnes BloodPressure, SkinThickness, Insulin, et BMI. Les laisser telles quelles fausserait toute analyse statistique ou modélisation.
Pour la colonne Pregnancies, la signification de zéro doit être comprise dans le contexte du domaine (femmes n'ayant jamais été enceintes versus données manquantes).

En résumé, le rapport de profilage met en évidence la propreté apparente du dataset en termes de valeurs manquantes (au sens NaN), mais alerte sur des valeurs numériques impossibles (les 0) qui devront être traitées comme des valeurs manquantes lors de la préparation des données. Il fournit également des informations cruciales sur les corrélations, en particulier celles liées à la variable Outcome.